auth-web/src/components/auth/account-settings/payment/AccountPaymentMethods.vue

```diff
@@ -103,269 +104,281 @@
-  private padValid: boolean = false
-  private eftValid: boolean = false
-  private paymentMethodChanged:boolean = false
-  private isFuturePaymentMethodAvailable: boolean = false // set true if in between 3 days cooling period
-  private isTOSandAcknowledgeCompleted:boolean = false // sert true if TOS already accepted
+  emits: ['emit-bcol-info'],
+  setup (props, { emit, root }) {
+    const orgStore = useOrgStore()
+    const userStore = useUserStore()
 
-  $refs: {
-      errorDialog: InstanceType<typeof ModalDialog>
-    }
+    const state = reactive({
+      savedOrganizationType: '',
+      selectedPaymentMethod: '',
+      padInfo: {} as PADInfo,
+      isBtnSaved: false,
+      disableSaveBtn: false,
+      errorMessage: '',
+      errorTitle: 'Payment Update Failed',
+      bcolInfo: {} as BcolProfile,
+      errorText: '',
+      isLoading: false,
+      padValid: false,
+      eftValid: false,
+      ejvValid: false,
+      paymentMethodChanged: false,
+      isFuturePaymentMethodAvailable: false, // set true if in between 3 days cooling period
+      isTOSandAcknowledgeCompleted: false // set true if TOS already accepted
+    })
 
-  private setSelectedPayment (payment) {
-    this.errorMessage = ''
-    this.selectedPaymentMethod = payment.selectedPaymentMethod
-    this.isBtnSaved = (this.isBtnSaved && !payment.isTouched) || false
-    this.paymentMethodChanged = payment.isTouched || false
-  }
+    const errorDialog = ref<InstanceType<typeof ModalDialog>>()
+
+    const currentOrganization = computed(() => orgStore.currentOrganization)
+    const currentOrgPaymentType = computed(() => orgStore.currentOrgPaymentType)
+    const currentMembership = computed(() => orgStore.currentMembership)
+    const currentOrgAddress = computed(() => orgStore.currentOrgAddress)
+    const permissions = computed(() => orgStore.permissions)
+    const currentUser = computed(() => userStore.currentUser)
 
-  private get isDisableSaveBtn () {
-    let disableSaveBtn = false
-    if (this.isBtnSaved) {
-      disableSaveBtn = false
-    } else if ((this.selectedPaymentMethod === PaymentTypes.PAD && !this.padValid) ||
-               (this.selectedPaymentMethod === PaymentTypes.EFT && !this.eftValid) ||
-               (!this.paymentMethodChanged) ||
-               (this.selectedPaymentMethod === PaymentTypes.EJV) ||
-               this.disableSaveButtonForBCOL()) {
-      disableSaveBtn = true
+    function setSelectedPayment (payment) {
+      state.errorMessage = ''
+      state.selectedPaymentMethod = payment.selectedPaymentMethod
+      state.isBtnSaved = (state.isBtnSaved && !payment.isTouched) || false
+      state.paymentMethodChanged = payment.isTouched || false
     }
 
-    return disableSaveBtn
-  }
+    const isDisableSaveBtn = computed(() => {
+      let disableSaveBtn = false
+      if (state.isBtnSaved) {
+        disableSaveBtn = false
+      } else if ((state.selectedPaymentMethod === PaymentTypes.PAD && !state.padValid) ||
```

Comment:
These conditionals should probably be a function?

Reply:
personally I'd break these down into payment types.. 

`isPadSelectedAndInvalid`
`isEftSelectedAndInvalid`
`isEjvSelectedAndInvalid`
`paymentMethodNotChangedAndEjvSelected`

Reply:
fixed